src/oneprovision/lib/provision.rb

```diff
@@ -490,7 +496,30 @@ def delete_object(type, id)
 
             Utils.exception(object.delete)
 
-            Utils.exception(object.wait_state('DONE')) if type == 'vm'
+            if type == 'vm'
+                Utils.exception(object.wait_state('DONE', timeout))
+            else
+                Utils.exception(wait_image_delete(object, timeout))
+            end
+        end
+
+        # Waits until the image is deleted
+        #
+        # @param image   [OpenNebula::Image] Image to wait
+        # @param timeout [Integer ]          Timeout for delete
+        def wait_image_delete(image, timeout)
+            timeout.times do
+                rc = image.info
+
+                if OpenNebula.is_error?(rc) && rc.message =~ /Error getting/
```

Comment:
I would say it's better to take any error as a successfull delete, instead of depending on some texts which might easily change in the future.